### ISSUE TITLE
Rulebook Update

### ISSUE THREAD

**Issue by ARTenshi** [RoboCupAtHome/RuleBook#711]
Hello,

some ideas to discuss for the rulebook. The ones that you think that can progress can have their own Issue. You can propose your own ideas (maybe during this week?)

1. Clean Up. The objects to be cleaned should belong to the designated room. In theory, as the room designation is random, the robot, in principle, is able to place any object in any deposit in any room. Therefore, taking an object from the bedroom to the kitchen doesn't demonstrate any skill and it's time-consuming in the (rather arbitrary) 5 min frame.

2. Storing groceries. In the DSPL, I would like to propose a maximum height in the kitchen's cupboard/shelf. This year the upper level was too high and the robot had to perform unnatural motions that made it unstable (the torso and manipulator had to be placed to their maximum height and very close to the cupboard resulting in a poorly balanced robot).

3. Dishwasher Challenge. The objects on the table should be in a messy order and not only dishes. The robot should be able to recognise what to take to the dishwasher and what not to (extra points if it take those object to the trash bin). More object from one category can be placed.

4. HRI. The robot should specify what to take and where it is. For example, in the Dishwasher Challenge, the robot might just get close to the table and ordered "give me a spoon in three seconds" and then closed its gripper. No feedback to the user about where or which item to take neither confirmation from the robot if it has received the correct item; also, no HRI, the robot asked to place the item in its gripper and wait until it got closed, it didn't track the user's hand position. We are very close to the robot commanding from the entrance to the kitchen: Take the spoon on the table and place it in the dishwasher, you have five seconds.

Besides, we should highlight the Award that is given to the best HRI system and that it is not limited to the hardware in the robot (I have seen a couple of open demos where they use Telegram to communicate with the user, seeing that applied in a challenge in at Home will be very interesting).

5. Positive point-scoring. I would like to propose that the robots sum points instead of decreasing them. If they perform the whole task within 5 mins, they get, say, 200 points; if it is in 4 mins or less, 100 extra points, if it is after 5 mins, they get only half of the total points. And if the intermediate tasks are performed correctly, they sum points, for example, the correct object moved to correct location, 100 points per object, correct drink (100 points) to correct person (100 points) sum 200 points, and so on. This year there were many 0 points teams but not because they didn't perform anything, but because of the point system. A positive point-scoring system might encourage teams while helping to asses the current state of the robots.

Regards,
Luis

**Comment by kyordhel** [comment]
We should split this into several issues, each targeting a specific task.

**Comment by ARTenshi** [comment]
Let's see first which ideas have chances to progress...

**Comment by semaph0r** [comment]
# 1 Clean Up

I think we can do this for the upcoming rulebook. But we should keep it in mind to change it later. Humans are a mess, to have a more realistic setting I want to see progression in handling dynamic environments. Therefore my proposal is to encourage teams to develop behaviour that enables robots to handle cleanup tasks en passant.

# 2 Storing groceries

I'm not experienced with the DSPL robots, but this sounds reasonable.

# 3 Dishwasher challenge

If there is a messy order, we need to emphasize the number and kind of items to be handled for obtaining all points. @ARTenshi What is your scoring scheme regarding trash disposal as you mentioned a encouraging scoring previously?

We should also be aware of not making this single task to complex as there are many objectives that are not yet fully covered by the teams.

# 4 HRI

Not sure what you mean with this. I assume that you want to prevent extensive rulebook exploitation. As of now, a robot has to specify the item which needs to be handed by the operator. This needs to be as precise as possible with gestures or with voice.

Regarding HRI Award: I will definetly support that. A robot at home is just another IOT-thing. In smart home environments we can leverage HRI approaches by the adoption of smart home integrations. Teams just need to think out of the box (manly out of robot).

# 5 Positive scoring

I opt-in for that. Rewarding is always better than punishment. It's a motivator.

**Comment by ARTenshi** [comment]
1. Clean Up

"Therefore my proposal is to encourage teams to develop behaviour that enables robots to handle cleanup tasks en passant." I agree, but to do so we need to objectively consider the timing. We either increase the time so the robot can clean an entire house (moving objects from one room to another) or with a short time cleaning a single room. Maybe for first stage, a short time is OK and for second stage, larger time is recommended inside an EGPSR task. 

So, my recommendation is to clarify that for Clean Up (First Stage) the objects will belong to the same room while for EGPSR they may belong to any room.

2 Storing groceries

Cool, we can change that. I will ask Toyota people what is the maximum height recommended for safe operation.

3 Dishwasher challenge

"What is your scoring scheme regarding trash disposal." I was thinking about the robot being able to recognise the (at least five) objects in any configuration and only take them to complete the task. If they move extra objects to the dishwasher or the garbage to the trash can, it's considered bonus points. We can actually see what's the average number of objects per minute a robot can successfully move for future regulations.

4. HRI

"This needs to be as precise as possible with gestures or with voice." Exactly. We should highlight what's a minimum expected HRI (e.g. an image, the robot pointing to the object or describing it by voice, the robot recognising the user's hand). We can discuss those minimums.

5 Positive scoring

Cool. I will work on a scoring draft for a couple of tasks to see if it makes sense.

**Comment by johaq** [comment]
1. Clean up. I don't fully get this point. Object to room designation is not random. It is assigned by TC/OC during setup days. The robot cannot just place any object in any deposit it has to bring the object to their predefined location as the rules currently state.

2. 95% agree on this. Of course there have to be enough objects that can be placed in a reasonable manner for the HSR. I still think that it is fine if there are objects out of reach. This just means the robot has to make smart choices by picking up objects that belong to categories that are in reach. This is something we also expect from OPL robots. This imo does not require a rule change just a heads up to TC to setup the task correctly.

3. I don't like the trash part since that is already part of other tasks (clean up). I like the messy order because it fits more with the setup of the task (table after family dinner). I would stick with only cutlery on the table for now since I don't think performance in this task was good enough to make it significantly harder.

4. The scenario that you describe should be covered by 3.8.2 (No awareness). It should carry the score reduction penalty.

5. This is simply not possible due to the competition schedule. We generally cannot let teams run over time for reduced scores. Furthermore I would like to advocate in favor of time limits. After all @home is a competition. An approaching time limit can be one of the most exciting parts (and frustrating admittedly) for teams and for the audience as well as it is a very easy to understand aspect of an otherwise very complicated contest.

**Comment by ARTenshi** [comment]
1. Yes, they have. And therefore placing objects that belong to a specific room in a messy order should be easy. Placing objects that belong to different rooms should be no problem either (as the room is randomly assigned, the robot should be able to clean any object in any room). However, this point is more about irreducible time-consuming mid-steps (like navigating from one room to another, if necessary). This irreducible time-consumption might prevent any team to fully perform the task regardless of the strategy they use (even with HRI, asking a human to take an object pointed by the robot and guiding the human to the correct location won't prevent the navigation time consumption between rooms).

In principle, as the room is designated randomly, we can consider as granted that a robot can place any object in any correct place and we should focus on time-realistic setups (as having objects that belong to the room in the room) and fully ordering objects inside the designated room should be fine.

2. Sure. It's more about depositing them. It was a deposit and was too high for safe manipulation.

3. Ok, I agree with that.

4. It is, but it wasn't performed like that. I just want to highlight that next year referees won't respond to weak interactions (like "give me a spoon, put it in my gripped and wait until I close it").

5. I agree with that, I just don't like the arbitrary time designation and would like to have at least a couple of competitions with this relaxed time window to see what's the real average time performance per task and then fix it for future competitions. With very shot times, teams rely on humans to perform many of the actions that may result in excessive use of HRI and it doesn't reflect the real potential of the robots.

**Comment by johaq** [comment]
1. If you take long for navigation and hri you are gonna score less points than a fast robot. It is 100 points per object so even if you don't manage all 5 you'll still score. Additionally, I wanna reward teams that put in the effort and have their robot pick in a smart way. Maybe a team has the robot calculate how long the drives would be to clean up two objects and pick the shorter one.
4. Yeah, a referee handbook would still be a nice to have...
5. With the schedule as is all tasks in the same stage and scenario need the same time limit. So we'd have to adjust the tasks if the time limit is way off. Was there a lot of running out of time in sydney that actually cost points? In my experience often time runs out for robots only standing around or driving around aimlessly.

**Comment by kyordhel** [comment]
Guys, I'm loving this discussion so far and specially the enthusiasm, yet I have to give some pinpoints.

1. Please open an issue per test, otherwise it becomes hard to follow and stay on topic
2. Try to keep tests within the 5min time limit. I don't want to elaborate on this here (off-topic), so if you have any questions, ask me directly.

**Comment by ARTenshi** [comment]
Some consensus can be quickly made here before raising many issues (e.g. Storing groceries maximum height, Dish washer messy order and no extra garbage, and clear and more precise/specific HRI). Maybe Clean Up can be agreed here before rising an issue? Positive scoring needs an issue or goes to roadmap?

**Comment by semaph0r** [comment]
Positive scoring needs to be an item on the roadmap at least. I'm not sure how much effort it takes to get a consensus. I'll pick one task and prototype a score scheme for that particular task to discuss general things on it.

I think Clean up is done here.
Also Storing groceries, dishwasher task and HRI

**Comment by ARTenshi** [comment]
What's the outcome? I see the two options about objects category in a room belongs to the room, rooms are designated aleatory. Objects in a room belongs to any category, rooms are designated aleatory. 

The second is ideal but the first is more realistic regardless the strategy.

**Comment by kyordhel** [comment]
> Positive scoring needs to be an item on the roadmap at least

Positive scoring is not subject to discussion. Penalties are confusing and hard to score. Scoresheets must be designed to grant points for what the robot accomplishes, not deducting from what went wrong. 

Special case is DEM. I'm not big fan of it and I'm open to suggestions. Trustees want robots finishing tests and looking useful, not getting stuck and failing for small details so we can give them a push. However, a robot needing pushes can't score as high as a robot that is fully autonomous.

**Comment by johaq** [comment]
@ARTenshi are your concerns still there with changes in #722? Can this be closed?

**Comment by ARTenshi** [comment]
Yes, thank you.
